Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift

```diff
@@ -603,6 +618,9 @@ extension NIOThrowingAsyncSequenceProducer {
                 failure: Failure?
             )
 
+            /// The state once a call to next has been cancelled. To reach this state we must cancel the source.
```

Comment:
This comment is a bit confused: presumably we're cancelling the iterator, not the source.

Reply:
```
            /// The state once a call to next has been cancelled. Cancel the source when entering this state.
```
@Lukasa Is this better?

Reply:
Yeah, LGTM.